src/server.ts

```diff
@@ -141,43 +147,107 @@ export class Server {
    *   console.log("latestLedger:", data.latestLedger);
    * });
    *
-   * Allows you to directly inspect the current state of a contract. This is a backup way to access your contract data which may not be available via events or simulateTransaction.
+   * Allows you to directly inspect the current state of a contract. This is a
+   * backup way to access your contract data which may not be available via
+   * events or simulateTransaction.
    *
-   * @param {string} contractId - The contract ID containing the data to load. Encoded as Stellar Contract Address e.g. `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex string for backwards compatibility, but will likely be deprecated in the future.
+   * By default, we will request both the 'temporary' and 'persistent' versions
+   * of the ledger durability. Only one should exist.
+   *
+   * @param {string} contractId - The contract ID containing the data to load.
+   *    Encoded as Stellar Contract Address e.g.
+   *    `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex
+   *    string for backwards compatibility, but will likely be deprecated in the
+   *    future.
    * @param {xdr.ScVal} key - The key of the contract data to load.
-   * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the {@link SorobanRpc.LedgerEntryResult} object with the current value.
+   * @param {string} [expirationType] - The "durability keyspace" that this
+   *    ledger key belongs to. By default, it tries requesting both the
+   *    'temporary' and 'persistent' versions so that you don't need to track
+   *    this separately.
+   *
+   *    WARNING: In the default case, only one should *actually* exist! If BOTH
+   *    exist, this function will throw as it cannot distinguish between them.
+   *
+   * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the
+   *    {@link SorobanRpc.LedgerEntryResult} object with the current value.
+   *
+   * @warning If the data entry in question is a 'temporary' entry, it's
+   *    entirely possible that it has expired out of existence. Future versions
+   *    of this client may provide abstractions to handle that.
    */
   public async getContractData(
     contractId: string,
     key: xdr.ScVal,
+    expirationType?: 'temporary' | 'persistent'
```

Comment:
A few thoughts 
1. Should we remove the optionality from the storage type? This feels unnecessarily magical and we're introducing more than one way to do something very simple. When a developer queries storage directly it assumes a familiarity with the storage layout of the contract and so it's expected that they know which type of storage is used. 
2. Should we change the api to more closely resemble the SDK? Often times it's the same developer(s) writing the contract and dapps so similar is better (to the extent possible). In the SDK we made a conscious decision to have clear namespace separation for persistent and temporary storage. example `env.storage().persistent().get()`. Maybe in this case just have two separate functions?
3. On the naming front, a few sdk iterations ago we replaced `data` with `storage`. Should we use `storage` here too? to complicate things further the json-rpc method is called `getLedgerEntries` so we have the same thing called three different names across three different layers of the stack - what a party :)

Reply:
> Should we change the api to more closely resemble the SDK

Probably no. Given that this is the `Server` object I think we should be as close as possible to the json-rpc API, not the SDK. In that case, should we change it to be `getLedgerEntries(keys)` similar to the server? We can introduce helpers elsewhere for creating keys.    

(Whether the sdk and rpc should be more similar is a different question I guess)

Reply:
Yeah I'm certainly not opposed to durability being explicit (discussed in [Slack thread](https://stellarfoundation.slack.com/archives/C02B04RMK/p1687986170249409)). Two separate functions seems like overkill, but you make a good point about developers having experience transfer from contract to client. I changed the behavior in ec67a12, wdyt?

For your latter points, there already **is** a `Server.getLedgerEntries(keys: xdr.LedgerKey[])`, so there should not be much confusion. This method, in fact, is intended to be a helper oriented around fetching `LedgerKeyContractData`s, hence the name (over "storage") and attempt at keeping it easy to use.

Reply:
oh cool. looks good and thanks for the clarification!